contracts/GovNftSugar.vy

```diff
@@ -52,121 +62,149 @@ interface IGovNFT:
   def locks(_govnft_id: uint256) -> Lock: view
   def unclaimed(_govnft_id: uint256) -> uint256: view
   def locked(_govnft_id: uint256) -> uint256: view
+  def name() -> String[100]: view
+  def symbol() -> String[100]: view
+  def owner() -> address: view
 
 interface IToken:
   def delegates(_account: address) -> address: view
 
 # Vars
-nft: public(IGovNFT)
+factory: public(IGovNFTFactory)
+collections: public(DynArray[address, MAX_COLLECTIONS])
 
 # Methods
 
 @external
-def __init__(_nft: address):
+def __init__(_factory: address):
   """
-  @dev Set up the GovNFT contract
+  @dev Set up the GovNFTFactory contract
   """
-  self.nft = IGovNFT(_nft)
+  self.factory = IGovNFTFactory(_factory)
+  self.collections = self.factory.govNFTs()
 
 @external
 @view
-def owned(_account: address) -> DynArray[GovNft, MAX_RESULTS]:
+def all_collections() -> DynArray[Collection, MAX_COLLECTIONS]:
```

Comment:
Would consider dropping `all` from `all_collections`, also a docstring would be really helpful